### ISSUE TITLE
Gatsby build crashes with JS heap out of memory when using CSS property "content"

### ISSUE THREAD

**Issue by gorohoroh** [gatsbyjs/gatsby#36021]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Running `gatsby build` fails with OOM ("JS heap out of memory") when building production bundles in a project where:
1. The content of an HTML element in a page is a string with at least a pair of multiline comments (`/*`, `*/`)
2. The HTML element is styled using pseudoselectors where the `content` property has multiline comment delimiters as values:
    ```css
      .comment:before {
          content: "/* "
      }
      
      .comment:after {
          content: " */"
      }
    ```

Running `gatsby develop` works as expected.

Here's the full `gatsby build` log:

```shell
sers/gorohoroh/.nvm/versions/node/v16.13.1/bin/node /Users/gorohoroh/.nvm/versions/node/v16.13.1/lib/node_modules/npm/bin/npm-cli.js run build --scripts-prepend-node-path=auto

> gatsby-starter-hello-world@0.1.0 build
> gatsby build

success compile gatsby files - 0.550s
success load gatsby config - 0.009s
success load plugins - 0.293s
success onPreInit - 0.001s
success initialize cache - 0.082s
success copy gatsby files - 0.168s
success Compiling Gatsby Functions - 0.357s
success onPreBootstrap - 0.378s
success createSchemaCustomization - 0.003s
success Checking for changed pages - 0.005s
success source and transform nodes - 0.067s
info Writing GraphQL type definitions to /Users/gorohoroh/SOURCE/gatsby-test/my-hello-world-starter/.cache/schema.gql
success building schema - 0.246s
success createPages - 0.003s
success createPagesStatefully - 0.058s
info Total nodes: 21, SitePage nodes: 1 (use --verbose for breakdown)
success Checking for changed pages - 0.001s
success onPreExtractQueries - 0.001s
success extract queries from components - 0.064s
success write out redirect data - 0.004s
success onPostBootstrap - 0.001s
info bootstrap finished - 6.991s
success write out requires - 0.038s
⠋ Building production JavaScript and CSS bundles
/Users/gorohoroh/SOURCE/gatsby-test/my-hello-world-starter/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

Error: Worker terminated due to reaching memory limit: JS heap out of memory
    at new NodeError (node:internal/errors:371:5)
    at Worker.[kOnExit] (node:internal/worker:276:26)
    at Worker.<computed>.onexit (node:internal/worker:198:20)
Emitted 'error' event on Worker instance at:
    at Worker.[kOnExit] (node:internal/worker:276:12)
    at Worker.<computed>.onexit (node:internal/worker:198:20) {
  code: 'ERR_WORKER_OUT_OF_MEMORY'
```

### Reproduction Link

https://github.com/gorohoroh/gatsby-build-oom-repro

### Steps to Reproduce

1. Clone https://github.com/gorohoroh/gatsby-build-oom-repro
2. Run `gatsby build`

### Expected Result

The production build process ran to completion.

### Actual Result

Process crash due to JS heap OOM

### Environment

```shell
System:
    OS: macOS 11.4
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 103.0.5060.53
    Firefox: 101.0.1
    Safari: 14.1.1
  npmPackages:
    gatsby: ^4.17.2 => 4.17.2 
  npmGlobalPackages:
    gatsby-cli: 4.13.0
```

**Comment by tyhopp** [comment]
Hi @gorohoroh, thanks for the detailed description and reproduction. Can you share more about what your use case is?

Escaping the characters like this renders what you would expect in the DOM and does not break when building:

```css
.comment:before {
  content: "\/\* ";
}

.comment:after {
  content: " \*\/";
}
```

**Comment by gorohoroh** [comment]
Ping @tyhopp, this issue is not awaiting author response anymore, see above.

**Comment by tyhopp** [comment]
Hi @gorohoroh, I'm sorry you had to experience that.

As for the ask here, I agree it would have been better if you were able to get a more precise error message. However given that you are unblocked by the change I suggested, this is very much an edge case, and the error technically isn't correct if the machine did run out of memory during the JS/CSS bundling step, I can't prioritize this at the moment.

If you (or anyone else reading) is interested in pursuing this further I'd start by looking at the CSS pipeline in webpack (see https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/webpack-utils.ts) and attempt to reproduce this outside of Gatsby with just webpack. From there you would be able to see if the issue is with a particular plugin (e.g. `css-loader` or `css-minimizer-webpack-plugin`), and if it is then see if an upstream fix can be made if appropriate.
